File: src/types/valorant.ts
// Valorant TypeScript types

export interface ValorantItem {
  name: string;
  price: number; // VP
  image_url: string | null;
  item_type: string; // e.g., "skin", "bundle"
  uuid?: string; // Item UUID
}

export interface ValorantStore {
  date: string; // ISO date
  items: ValorantItem[];
  checked_at: string; // ISO timestamp
  is_real_data?: boolean; // Whether this is real API data or mock data
}

export interface GetStoreHistoryParams {
  limit: number | null;
}

// Authentication status
export interface AuthStatus {
  is_authenticated: boolean;
  has_full_cookies: boolean;
  username: string | null;
  region: string;
  puuid: string | null;
  expires_hint: string | null; // "3 weeks" or "1 week"
}

// Event payloads
export interface ValorantUpdatedEvent {
  store: ValorantStore;
}

File: src-tauri/src/models/valorant.rs
// Valorant data models
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ValorantItem {
    pub name: String,
    pub price: u32,
    pub image_url: Option<String>,
    pub item_type: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ValorantStore {
    pub date: String,
    pub items: Vec<ValorantItem>,
    pub checked_at: String,
    #[serde(default)]
    pub is_real_data: Option<bool>,
}

impl ValorantStore {
    pub fn new(items: Vec<ValorantItem>) -> Self {
        let now = chrono::Utc::now();
        Self {
            date: now.format("%Y-%m-%d").to_string(),
            items,
            checked_at: now.to_rfc3339(),
            is_real_data: None,
        }
    }
}
